lib/Elevate/Components/BootKernel.pm

```diff
@@ -82,4 +89,12 @@ sub _to_str ($e) {
     return $str;
 }
 
+sub pre_distro_upgrade ($self) {
```

Comment:
Instead of doing this sort of thing with `pre_distro_upgrade`, `post_distro_upgrade`, and `check` in individual components, perhaps these should be added to `Elevate::Components::Base` as default methods? Alternatively, if we think that there's value in having every subclass which doesn't use one of these explicitly override it with an empty method, we could add them to the base class as abstract methods which `die` if called without being overriden.

Reply:
I knew I had gotten too close to this.  Adding this to the base class and getting rid of the overrides.